modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/TxFinishReplicaRequestHandler.java

```diff
@@ -146,18 +145,18 @@ public CompletableFuture<TransactionResult> handle(TxFinishReplicaRequest reques
         }
     }
 
-    private static Map<TablePartitionId, String> asTablePartitionIdStringMap(Map<TablePartitionIdMessage, String> messages) {
-        var result = new HashMap<TablePartitionId, String>(IgniteUtils.capacity(messages.size()));
+    private static Map<ReplicationGroupId, String> asReplicationGroupIdToStringMap(Map<ReplicationGroupIdMessage, String> messages) {
+        var result = new HashMap<ReplicationGroupId, String>(IgniteUtils.capacity(messages.size()));
 
-        for (Entry<TablePartitionIdMessage, String> e : messages.entrySet()) {
-            result.put(e.getKey().asTablePartitionId(), e.getValue());
+        for (Entry<ReplicationGroupIdMessage, String> e : messages.entrySet()) {
+            result.put(e.getKey().asReplicationGroupId(), e.getValue());
         }
 
         return result;
     }
 
     private CompletableFuture<TransactionResult> finishAndCleanup(
-            Map<TablePartitionId, String> enlistedPartitions,
+            Map<ReplicationGroupId, String> enlistedPartitions,
```

Comment:
Are you going to add tableIds here while implementing [IGNITE-24384]? I mean that in order to switch writeIntents you will need to know which tables to touch.

Reply:
Yes, exactly